packages/@sanity/desk-tool/src/DeskTool.js

```diff
@@ -1,13 +1,36 @@
 import React from 'react'
 import styles from './styles/DeskTool.css'
 import SchemaPaneResolver from './SchemaPaneResolver'
+import client from 'part:@sanity/base/client'
+import {withRouterHOC} from 'part:@sanity/base/router'
+import PropTypes from 'prop-types'
+
+export default withRouterHOC(class DeskTool extends React.Component {
+  static propTypes = {
+    router: PropTypes.shape({
+      state: PropTypes.object
+    }).isRequired
+  }
+  componentDidMount() {
+    const {router} = this.props
+    const {selectedType, selectedDocumentId} = router.state
+    if (selectedDocumentId && selectedType === '*') {
+      client.fetch('*[_id == $id][0]{_type}', {id: selectedDocumentId}).then(res => res._type)
```

Comment:
Could be written as `*[_id == $id][0]._type`, perhaps?

Reply:
Ah, good point. Will fix.